packages/debugger/src/panels/variables/mimerenderer.ts

```diff
@@ -0,0 +1,65 @@
+import { MainAreaWidget } from '@jupyterlab/apputils';
+import { IRenderMimeRegistry, MimeModel } from '@jupyterlab/rendermime';
+import { PromiseDelegate } from '@lumino/coreutils';
+import { Panel } from '@lumino/widgets';
+import { IDebugger } from '../../tokens';
+
+/**
+ * Debugger variable mime type renderer
+ */
+export class VariableMimeRenderer extends MainAreaWidget<Panel> {
+  /**
+   * Instantiate a new VariableMimeRenderer.
+   */
+  constructor(options: VariableMimeRenderer.IOptions) {
+    const { dataLoader, rendermime } = options;
+    const content = new Panel();
+    const loaded = new PromiseDelegate<void>();
+    super({
+      content,
+      reveal: Promise.all([dataLoader, loaded.promise])
+    });
+
+    dataLoader
+      .then(async data => {
+        if (data.data) {
+          const mimeType = rendermime.preferredMimeType(data.data);
```

Comment:
`safe` parameter of `preferredMimeType` of need to be `'any'` in order to render correctly widgets. 
```suggestion
          const mimeType = rendermime.preferredMimeType(data.data, 'any');
```

Reply:
Thanks @trungleduc